### PR TITLE
[metrics] speed up polygon iou (for --rotation) by keeping balanced memory footprint

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- boost polygon iou (from hours to a few minutes) and try to keep an balanced memory consumption
- remove now unused functions in `metrics.py`
- remove unused `use_broadcasting` / `mask_shape` and `psutil` requirements for references
- memory usage tested with oversampled funsd validation dataset (~4K samples) tooks around 6.5 GB RAM

Any feedback is welcome :hugs: 

Closes: #1472

### CODE REVIEW

**Comment by felixdittrich92** [comment]
failing tests: hf hub is down :D

File: doctr/utils/metrics.py [hunk 1]
```diff
@@ -234,80 +173,18 @@ def polygon_iou(
...
+    shapely_polys_1 = [Polygon(poly) for poly in polys_1]
+    shapely_polys_2 = [Polygon(poly) for poly in polys_2]
 
-def _rbox_to_mask(box: np.ndarray, shape: Tuple[int, int]) -> np.ndarray:
-    """Converts a rotated bounding box to a boolean mask
+    for i, poly1 in enumerate(shapely_polys_1):
+        for j, poly2 in enumerate(shapely_polys_2):
+            intersection_area = poly1.intersection(poly2).area
+            union_area = poly1.area + poly2.area - intersection_area
+            iou_mat[i, j] = intersection_area / union_area
```
Comment (odulcy-mindee) [inline doctr/utils/metrics.py:185]: It's a symmetrical matrix so you can compute half the matrix and guess the rest

File: doctr/utils/metrics.py [hunk 2]
```diff
@@ -441,7 +312,7 @@ def summary(self) -> Tuple[Optional[float], Optional[float], Optional[float]]:
         precision = self.matches / self.num_preds if self.num_preds > 0 else None
 
         # mean IoU
-        mean_iou = self.tot_iou / self.num_preds if self.num_preds > 0 else None
+        mean_iou = round(self.tot_iou / self.num_preds, 2) if self.num_preds > 0 else None
```
Comment (odulcy-mindee) [inline doctr/utils/metrics.py:315]: why rounding ?

**Review by odulcy-mindee — CHANGES_REQUESTED** [review]

File: doctr/utils/metrics.py (hunk 2 above)
Comment (felixdittrich92) [reply doctr/utils/metrics.py:315]: Because there may be minimal calculation deviations like 0.12999999......8 instead of 0.13 for example

**Review by odulcy-mindee — APPROVED** [review]

**Pull request merged**